roles/ceph-grafana/tasks/configure_grafana.yml

```diff
@@ -6,8 +6,8 @@
   register: result
   until: result is succeeded
   when:
-    - not containerized_deployment
```

Comment:
Shouldn't we keep this because doing a contianerized deployment, to me, means not installing packages?

Reply:
The _ceph-grafana-dashboards_ package provides a set of yaml files that Grafana uses as the default dashboard layout [1].
Removing this boolean we need to update the grafana container config [2] or, if the package is not present, the container will fail with the following:

```
lvl=eror msg="server shutdown" logger=server reason="service init failed: failed to provision dashboards: failed to provision config ceph dashboard: stat /etc/grafana/dashboards/ceph-dashboard: no such file or directory"
```

This happens because, as the systemd unit shows [3], we mount the whole /etc/grafana to properly configure the container: without this volume it will result in a classic grafana-server without the dashboard layouts  but with the prometheus datasource configured, (in this case, the operator should  configure the cluster).
I suggest to decouple the grafana bootstrapping phase with the grafana config phase, so we can better handle how this package is provided and what kind of config we need to pass to the grafana server.
Before update the PR, i would like to collect some considerations.

@fultonj , @guits , @dsavineau , what do you think?


[1] https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-defaults/defaults/main.yml#L724
[2] https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-grafana/templates/dashboards-ceph-dashboard.yml.j2#L12
[3] https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-grafana/templates/grafana-server.service.j2#L16

Reply:
@fmount see https://github.com/ceph/ceph/pull/26415